<!-- (c) 2019 PMD -->
Using a branching statement as the last part of a loop may be a bug, and/or is confusing. Ensure that the usage is not a bug, or consider using another approach.

<h2>Noncompliant Code Example</h2>
<pre>
// unusual use of branching statement in a loop
for (int i = 0; i < 10; i++) {
  if (i*i <= 25) {
    continue;
  }
  break;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
// this makes more sense...
for (int i = 0; i < 10; i++) {
  if (i*i > 25) {
    break;
  }
}
</pre>
